src/lib/rank_utils.ts

```diff
@@ -22,9 +22,74 @@ interface IRankInfo {
     label: string;
 }
 
-export const MaxRank: number = 36;
+class Rating {
+    unset:boolean;
+    rating:number;
+    deviation:number;
+    volatility:number;
+    rank:number;
+    rank_label:string;
+    partial_rank:number;
+    partial_rank_label:string;
+    rank_deviation_labels:Array<string>;
+    rank_deviation:number;
+}
+
+export const MaxRank: number = 39;
+
+
+const MIN_RATING = 100;
+const MAX_RATING = 6000;
+
+export function rank_to_rating(rank:number) {
+    return 900 * Math.exp(0.032 * rank);
+}
+
+export function rating_to_rank(rating:number) {
+    return Math.log(Math.min(MAX_RATING, Math.max(MIN_RATING, rating)) / 900.0) / 0.032;
+}
+
+export function get_handicap_adjustment(rating:number, handicap:number):number {
+    return rank_to_rating(rating_to_rank(rating) + handicap) - rating;
```

Comment:
Some discussion around this would be very beneficial.

I had glossed over this because my mind immediately went to "Oh that's just a compensation for the fact that handicaps are going to be most commonly determined by taking the difference in an rank, with the rank being considered as a whole number. Since we'll be computing handicaps using fractional rankings, we shouldn't do that".

But now thinking again, I'm not sure that was their reasoning for deducting `0.5`, since the case where you have a 1 handicap game between a two people rated `1299` and `1301` (so really the handicap should probably be 0) is expected to be as common as a 1 handicap game between two people rated `1201` and `1399` (so really the handicap should probably be 2). So now I'm thinking that `0.5`probably has nothing to do with that.

Do you happen to know what they're trying to account for with that adjustment?